Student-Service/src/main/java/tw/waterball/judgegirl/studentservice/domain/usecases/DeleteGroupByIdUseCase.java

```diff
@@ -0,0 +1,29 @@
+package tw.waterball.judgegirl.studentservice.domain.usecases;
+
+import lombok.AllArgsConstructor;
+import tw.waterball.judgegirl.commons.exceptions.NotFoundException;
+import tw.waterball.judgegirl.studentservice.domain.repositories.GroupRepository;
+
+import javax.inject.Named;
+
+/**
+ * @author - wally55077@gmail.com
+ */
+@Named
+@AllArgsConstructor
+public class DeleteGroupByIdUseCase {
+
+    private final GroupRepository groupRepository;
+
+    public void execute(Integer groupId) throws NotFoundException {
+        validateGroupIdExist(groupId);
+        groupRepository.deleteGroupById(groupId);
+    }
+
+    private void validateGroupIdExist(Integer id) throws NotFoundException {
+        if (!groupRepository.existsById(id)) {
+            throw new NotFoundException();
+        }
+    }
```

Comment:
Why don't you throw `EmptyResultDataAccessException` ?
Then the code will look as following:
```java
public void execute(Integer groupId) throws NotFoundException {
        try {
              groupRepository.deleteGroupById(groupId);
        } catch (EmptyResultDataAccessException e) {
              throw new NotFoundException();
        }
    }
```